lib/std/io/auto_indenting_stream.zig

```diff
@@ -0,0 +1,150 @@
+const std = @import("../std.zig");
+const io = std.io;
+const mem = std.mem;
+const Allocator = mem.Allocator;
+const ArrayList = std.ArrayList;
+const assert = std.debug.assert;
+
+/// Automatically inserts indentation of written data by keeping
+/// track of the current indentation level
+pub fn AutoIndentingStream(comptime indent_delta: usize, comptime WriterType: type) type {
```

Comment:
Any reason for `indent_delta` to be comptime known? We can reduce code bloat by making this initialized at runtime if possible.

Reply:
No special reason, it was just a comptime value prior to the indenting stream. Will change.